### PR TITLE
Created test for mail and added 'To' Field

### PR DESCRIPTION
**Pull request by sjainit**
Please review @paramsingh.

### CODE REVIEW

File: .cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (paramsingh) [inline .cache/v/cache/lastfailed]: This file isn't needed.

File: brainzutils/test/test_mail.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import smtplib
+
+from mock import patch, call
+from brainzutils import flask
+from brainzutils import mail
+
+
+class SendEmailTests(unittest.TestCase):
+
+    @patch('brainzutils.mail.send_mail')
```
Comment (paramsingh) [inline brainzutils/test/test_mail.py]: If you mock the brainzutils send mail function, then there is no point of this test. You should try to mock out the function that generates the smtp object, hence getting a MagicMock object in place of the smtp object and then check that the smtp send_mail function is called correctly.

**Comment by sjainit** [comment]
@paramsingh, to resolve the assertion error, I had to create the message in the way as our function creates it. Also, while creating the message the boundary parameter was not fixed, so it varied each time, due to which I had to set the boundary parameter while testing. 
The build is still failing due to the same error as before, i.e. `cannot import current_app from flask` in mail.py.

File: .cache/v/cache/lastfailed (hunk 1 above)
Comment (sjainit) [reply .cache/v/cache/lastfailed]: @paramsingh, I did not create it. I don't really know how it got created and can't delete as I don't know its location where it got created.

**Comment by sjainit** [comment]
@paramsingh, the tests have passed. :)

File: brainzutils/mail.py [hunk 3]
```diff
@@ -26,14 +29,20 @@ def send_mail(subject, text, recipients, attachments=None,
...
-
+              
     if current_app.config['TESTING']:  # Not sending any emails during the testing process
         return
 
     if not recipients:
         return
 
-    message = MIMEMultipart('mixed')
+    if test:
```
[the remark this answers is not in the document]
Comment (sjainit) [reply brainzutils/mail.py]: Oops, yup we can pass the `boundary` parameter. For the question, `mixed` is the `subtype` of the message we are sending. The default value of this parameter is 'mixed', so we can also not pass any value to the constructor and it should work fine.

File: brainzutils/test/test_mail.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            message = MIMEMultipart(boundary=originalboundary)
+            message['To']= '<%s>' %(recipients)
+            message['Subject'] = subject
+            message['From'] = "%s <%s>" % (from_name, from_address)
+            message.attach(MIMEText(text, _charset='utf-8'))
+
+            for attachment in attachments:
+                file_obj, subtype, name = attachment
+                attachment = MIMEApplication(file_obj.read(), _subtype=subtype)
+                file_obj.close()  # FIXME(roman): This feels kind of hacky. Maybe there's a better way?
```
[the remark this answers is not in the document]
Comment (sjainit) [reply brainzutils/test/test_mail.py]: Yes, sure will take care :)

File: brainzutils/test/test_mail.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            from_name='ListenBrainz'
+            subject='ListenBrainz Spotify Importer Error'
+            originalboundary = "===============2220963697271485568=="
+            message = MIMEMultipart(boundary=originalboundary)
+            message['To']= '<%s>' %(recipients)
+            message['Subject'] = subject
+            message['From'] = "%s <%s>" % (from_name, from_address)
+            message.attach(MIMEText(text, _charset='utf-8'))
+
+            for attachment in attachments:
```
[the remark this answers is not in the document]
Comment (sjainit) [reply brainzutils/test/test_mail.py]: Yes, right. I have also made changes to `mail.py` file accordingly.

File: brainzutils/test/test_mail.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                file_obj, subtype, name = attachment
+                attachment = MIMEApplication(file_obj.read(), _subtype=subtype)
+                file_obj.close()  # FIXME(roman): This feels kind of hacky. Maybe there's a better way?
+                attachment.add_header('content-disposition', 'attachment', filename=name)
+                message.attach(attachment)
+
+            mail.send_mail(
+            subject='ListenBrainz Spotify Importer Error',
```
[the remark this answers is not in the document]
Comment (sjainit) [reply brainzutils/test/test_mail.py]: Yes, sure :)

File: brainzutils/test/test_mail.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import mock as mock
+
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from brainzutils import flask
+from brainzutils import mail
+
+
+class SendEmailTests(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (sjainit) [reply brainzutils/test/test_mail.py]: Makes sense.

**Comment by sjainit** [comment]
@paramsingh, can we upgrade the version of Flask and Mock? Should I do different PR for this or commit the changes here only?

File: brainzutils/mail.py [hunk 8]
```diff
@@ -22,29 +23,35 @@ def send_mail(subject, text, recipients, attachments=None,
...
         return
 
     if not recipients:
         return
 
-    message = MIMEMultipart('mixed')
+    message =MIMEMultipart()
+
+    if boundary is not None:
+        originalboundary = "===============2220963697271485568=="
```
Comment (paramsingh) [inline brainzutils/mail.py]: I don't like this, in general please avoid using magic constants in the code. Each constant should be named.

you should just do 


```
if boundary is not None:
    message = MIMEMultipart(boundary=boundary)
else:
    # set message to default value here
```

File: brainzutils/mail.py [hunk 9]
```diff
@@ -22,29 +23,35 @@ def send_mail(subject, text, recipients, attachments=None,
...
-
+              
     if current_app.config['TESTING']:  # Not sending any emails during the testing process
         return
 
     if not recipients:
         return
 
-    message = MIMEMultipart('mixed')
+    message =MIMEMultipart()
```
Comment (paramsingh) [inline brainzutils/mail.py]: why'd you remove the `mixed` here?

File: brainzutils/mail.py [hunk 10]
```diff
@@ -22,29 +23,35 @@ def send_mail(subject, text, recipients, attachments=None,
         from_name: Name of the sender.
         from_addr: Email address of the sender.
     """
-    if attachments is None:
-        attachments = []
+
     if from_addr is None:
         from_addr = 'noreply@' + current_app.config['MAIL_FROM_DOMAIN']
-
+
```
Comment (paramsingh) [inline brainzutils/mail.py]: remove these spaces please.

File: brainzutils/mail.py [hunk 11]
```diff
@@ -22,29 +23,35 @@ def send_mail(subject, text, recipients, attachments=None,
         from_name: Name of the sender.
         from_addr: Email address of the sender.
     """
-    if attachments is None:
```
Comment (paramsingh) [inline brainzutils/mail.py]: why'd you change the attachments code in the functiion, I only asked for the changes in the test, let's revert them back as the old way is a bit cleaner and pythonic. And I don't want to change code that is hard to test a lot.

Comment (sjainit) [reply brainzutils/mail.py]: Cool :)

File: brainzutils/mail.py (hunk 9 above)
Comment (sjainit) [reply brainzutils/mail.py]: Discussed on IRC :)

File: brainzutils/mail.py (hunk 8 above)
Comment (sjainit) [reply brainzutils/mail.py]: Quite dumb of me. Thankss :)

**Pull request merged**